source/base/bounding_box.cc

```diff
@@ -25,7 +24,8 @@ bool BoundingBox<spacedim,Number>::point_inside (const Point<spacedim, Number> &
       //Bottom left-top right convention: the point is outside if it's smaller than the
       //first or bigger than the second boundary point
       //The bounding box is defined as a closed set
-      if ( p[i] < this->boundary_points.first[i] || this->boundary_points.second[i] < p[i])
+      if ( p[i] + std::numeric_limits<double>::epsilon() < this->boundary_points.first[i]
+           || this->boundary_points.second[i] + std::numeric_limits<double>::epsilon() < p[i])
```

Comment:
I think you want to use `std::numeric_limits<Number>` instead.

But this is the wrong formula: http://en.cppreference.com/w/cpp/types/numeric_limits/epsilon shows that epsilon is only useful *when you compare with objects of size 1*. If, however, you are doing quantum mechanics and all of your lengths are on the order of 1e-15, then epsilon is a *very large* tolerance!

I think you want to do something like `p[i] * (1. + std::numeric_limits<Number>::epsilon())` which has the correct length.